tools/gulp/docs/parseReactFile.js

```diff
@@ -0,0 +1,89 @@
+const dutil = require('../common/log-util');
+const gutil = require('gulp-util');
+const path = require('path');
+const reactDocgen = require('react-docgen');
+const reactDocgenHandlers = require('./react-docgen-handlers');
+const through = require('through2');
+
+/**
+ * A Gulp plugin that generates JSON objects from a stream of React
+ * component or example files. The outputted objects are then used
+ * for rendering propType documentation and JS markup examples
+ * @param {Object} options
+ * @param {String} rootPath - Root docs site path
+ */
+module.exports = function(options = {}, rootPath) {
+  const response = {};
+
+  return through.obj((file, encoding, cb) => {
+    try {
+      if (file.isNull()) return cb(null, file);
+
+      const data = file.path.match(/.example.(jsx|js)$/)
+        ? parseExample(file)
+        : parseComponent(file, rootPath);
+
+      // Assign the data to a unique property, based on the filename,
+      // so we can merge the JSON files in another stream
+      response[getNameAfter(options.nameAfter, file.path)] = data;
+    } catch (e) {
+      dutil.logError('react-docgen', e);
+      dutil.logData('react-docgen', file.path);
+    }
+
+    file.contents = Buffer.from(JSON.stringify(response));
+    file.path = gutil.replaceExtension(file.path, '.json');
+    return cb(null, file);
+  });
+};
+
+/**
+ * For an example file, we only need a reference to its uncompiled source code
+ * @param {Object} file
+ */
+function parseExample(file) {
```

Comment:
ℹ️ This method returns the source code that's shown in the "Code snippet" section:

![image](https://user-images.githubusercontent.com/371943/33784770-0806be24-dc30-11e7-8a68-444119307363.png)